go/grpcweb/wrapper.go

```diff
@@ -184,8 +184,7 @@ func (w *WrappedGrpcServer) IsGrpcWebRequest(req *http.Request) bool {
 //
 // You can control the CORS behaviour using `With*` options in the WrapServer function.
 func (w *WrappedGrpcServer) IsAcceptableGrpcCorsRequest(req *http.Request) bool {
-	accessControlHeaders := strings.ToLower(req.Header.Get("Access-Control-Request-Headers"))
-	if req.Method == http.MethodOptions && strings.Contains(accessControlHeaders, "x-grpc-web") {
+	if req.Method == http.MethodOptions && strings.HasPrefix(req.Header.Get("content-type"), grpcContentType) {
```

Comment:
sniff `content-type` header using identical code from earlier in this same file

Reply:
Hey @sgammon, I just dug into this code with @MarcusLongmuir. 

It's important to understand that the client has no control over the `Options` request (it is made automatically by the browser) - this is known as the preflight request.

It's also important to note that the `content-type` header of the Options request is not the same as the `content-type` of the request that triggered the preflight request.

We believe the correct remedy is to:

1. Change this conditional check to simply look for Options requests, ie:

```
if req.Method == http.MethodOptions {
``` 

2. Fast follow on #158 (preferably in the same release as this change).

Give this a shot and hopefully the browser tests will pass :)

Reply:
thank you @jonny-improbable, i am aware of the distinction between the pre-flight and the `content-type` header here, but it's not my code, so i figured i would chip in to fix it because it breaks so frequently without custom CORS config

that being said, i was *not* aware the `content-type` isn't carried over in the pre-flight, which does make sense. of course there should be ways to avoid a pre-flight entirely, namely, omitting a custom `content-type` entirely (which breaks protocol spec but avoids an initial round-trip) or even leveraging browser form submission, but yes, we are on the same page.

in our environment unless i am mistaken we:
1) omit the `x-grpc-web` header to avoid the `Access-Control-Allow-Headers` change it requires
2) omit the `content-type`, affix it server-side before dispatching via gRPC

between those two i *believe* we avoid the pre-flight but i'll check to make sure. perhaps it might be a good option to add to improbable's branch of grpc-web since it seems you guys have control of the entire stack and could make a decision against spec like that, if you wanted. but hiding it behind a flag would be good :)

i'll go ahead and apply the conditional change, i am completely unfamiliar with golang but i will take your word for it